### PR TITLE
Add support for ZADD GT/LT options

### PR DESCRIPTION
**Pull request by dengliming**
Make sure that:

- [x] You have read the [contribution guidelines](https://github.com/lettuce-io/lettuce-core/blob/main/.github/CONTRIBUTING.md).
- [x] You have created a feature request first to discuss your contribution intent. Please reference the feature request ticket number in the pull request.
- [x] You use the code formatters provided [here](https://github.com/lettuce-io/lettuce-core/blob/main/formatting.xml) and have them applied to your changes. Don’t submit any formatting related changes.
- [x] You submit test cases (unit or integration tests) that back your changes.

 

closes [https://github.com/lettuce-io/lettuce-core/issues/1451](https://github.com/lettuce-io/lettuce-core/issues/1451)

### CODE REVIEW

File: src/main/java/io/lettuce/core/ZAddArgs.java [hunk 1]
```diff
@@ -109,6 +132,28 @@ public ZAddArgs ch() {
...
     }
 
+    /**
+     * Only update existing elements if the new score is greater than the current score. This flag doesn't prevent adding new elements.
+     *
+     * @return {@code this} {@link ZAddArgs}.
+     */
+    public ZAddArgs gt() {
+
+        this.gt = true;
```
Comment (mp911de) [inline src/main/java/io/lettuce/core/ZAddArgs.java:142]: ideally, we set `lt` to `false` to not have a state where `lt` and `gt` are both set to true if someone calls `lt().gt()`.

**Review by mp911de — CHANGES_REQUESTED** [review]
Thanks a lot, great work. Can you add also `@since` tags (version 6.1) to all newly introduced `public` methods in `ZAddArgs` and yourself as `@author`?

I think we also need a guard to the `@Test` methods when running the tests against an older Redis version, but I currently don't have a good idea. Maybe it's time to introduce `@EnabledOnRedisVersion`. Happy for suggestions.

**Comment by dengliming** [comment]
> Maybe it's time to introduce @EnabledOnRedisVersion.

@mp911de Great idea. Maybe we should track this with separated issue? Seems that we can get the current redis version by calling `RedisConditions.getRedisVersion`  WDYT?

**Comment by mp911de** [comment]
For now, we can guard the tests with `@EnabledOnCommand("ZMSCORE")` since `ZMSCORE` was added in Redis 6.2, too.

**Comment by dengliming** [comment]
Thanks. will update it later.

**Comment by mp911de** [comment]
No worries, I'm about to merge and polish up this PR.

**Comment by mp911de** [comment]
Thank you for your contribution. That's merged and polished now.

**Pull request closed without merging**